dspace-api/src/main/java/org/dspace/content/DSpaceObject.java

```diff
@@ -192,4 +202,70 @@ public void clearModified() {
     protected void setModified() {
         this.modified = true;
     }
+
+    public String toString() {
+        String ident = getHandle();
+        if (ident == null)
+            ident = "" + getID();
+        return Constants.typeText[getType()] + "." + ident;
+    }
+
+    /**
+     * expects a handle or a two part string as parameter: TYPE.IDENT,
+     * where TYPE is a DSpaceObject type and
+     * IDENT is a UUID, handle,  group name, or Eperson's netid
+     *
+     * returns DSpaceObject that corresponds to string or null
+     * @param str    must have format TYPE.ID
+     * @return
+     */
+    public static DSpaceObject fromString(Context c, String str) throws SQLException {
```

Comment:
this can't move to 

public interface DSpaceObjectService<T extends DSpaceObject> { ...} 

since fromString is a static method 

there does not seem to be a DSpaceObjectService<DspaceObject> class